File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        
        //shift all the elements to front
        int posNum = 0;
        
        for(int num : nums){
            if(num != 0) nums[posNum++] = num;
        }
        
        //fill the left elements to 0 
        while(posNum<nums.length){
            nums[posNum++] = 0;
        }
        
        
        
    }
}

File: 0283-move-zeroes/0283-move-zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
       
        let left = 0;
        
        for(let i=0; i<nums.length; i++){
            if(nums[i] !== 0){
                nums[left] = nums[i];
                left++;
            }
        }
    
        for(let i=left; i<nums.length; i++){
            nums[i] = 0;
        }
    
};